src/model/ParentObject.cpp

```diff
@@ -88,14 +88,16 @@ namespace detail {
 
     // subTree includes this object, make sure to include costs as well
     auto subTree = getRecursiveChildren(getObject<ParentObject>(), true);
-    // drop the Curve instances
+
+    // drop the Curve instances, if they are used by other objects
     // Perhaps this could be done in the getRecursiveChildren, but this way
     // the getRecursiveChildren method might be less surprising
     // This is probably the unique situation where you want to get children minus curves
-    auto isCurve = [](const ModelObject & modelObject) {
-      return modelObject.optionalCast<Curve>();
+    auto isUsedCurve = [](const ModelObject & modelObject) {
+      auto _curve = modelObject.optionalCast<Curve>();
+      return (_curve && _curve->directUseCount() > 1);
```

Comment:
Alternative:

```
auto isCurve = [this](const ModelObject & modelObject) {
   return modelObject.optionalCast<Curve>() && (modelObject.handle() != this->handle());
}
```

Or, just override remove and test in `Curve_Impl::remove`: cf https://github.com/NREL/OpenStudio/issues/3153#issuecomment-416371906

Reply:
Yeah. Why can't this just be implemented in Curve_Impl::remove? I don't care for the type checking in the base class. I know there was some already there before you did this @jmarrec, but maybe now is the time to clean it up.

Reply:
Curve_Impl::remove is never called if you are trying to remove a ParentObject. https://github.com/NREL/OpenStudio/blob/156dca45c0ed89ca4386bbe14cca530c11b9fbef/src/model/ParentObject.cpp#L80:L87

Reply:
Do you think that perhaps this filtering and behavior should happen for all ResourceObjects that are the child of a parent object? The fact that we are treating Curves in a special way concerns me a little bit.

Reply:
I'm not sure I'd want to delete a now-unused `SpaceLoadDefinition` object if I remove the parent `SpaceLoadInstance`. And neither remove a `SpaceLoadInstance` if I remove the `Space`/`SpaceType` attached to it. (I also though about constructions, but apparently they never get parented)

Thinking about it a bit more, I think the main difference between curves and the rest is probably stemming from two things:
* Primary: We instantiate a lot of default curves in constructors because they are required properties, especially HVAC/ZoneHVAC objects. So that ends up bloating the model a ton more than any other `ResourceObject`
* Secondary (and a bad justification probably): the OS App (still) has no way to edit, add, or remove Curve objects, so it's much harder to clean up stuff.

Reply:
ok well that seems like enough justification to me. I'll go ahead and merge this.